pkg/generated/controllers/management.cattle.io/v3/podsecurityadmissionconfigurationtemplate.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2022 Rancher Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by main. DO NOT EDIT.
+
+package v3
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/lasso/pkg/client"
+	"github.com/rancher/lasso/pkg/controller"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/generic"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+type PodSecurityAdmissionConfigurationTemplateHandler func(string, *v3.PodSecurityAdmissionConfigurationTemplate) (*v3.PodSecurityAdmissionConfigurationTemplate, error)
+
+type PodSecurityAdmissionConfigurationTemplateController interface {
+	generic.ControllerMeta
+	PodSecurityAdmissionConfigurationTemplateClient
+
+	OnChange(ctx context.Context, name string, sync PodSecurityAdmissionConfigurationTemplateHandler)
+	OnRemove(ctx context.Context, name string, sync PodSecurityAdmissionConfigurationTemplateHandler)
+	Enqueue(namespace, name string)
+	EnqueueAfter(namespace, name string, duration time.Duration)
+
+	Cache() PodSecurityAdmissionConfigurationTemplateCache
+}
+
+type PodSecurityAdmissionConfigurationTemplateClient interface {
+	Create(*v3.PodSecurityAdmissionConfigurationTemplate) (*v3.PodSecurityAdmissionConfigurationTemplate, error)
+	Update(*v3.PodSecurityAdmissionConfigurationTemplate) (*v3.PodSecurityAdmissionConfigurationTemplate, error)
+
+	Delete(namespace, name string, options *metav1.DeleteOptions) error
+	Get(namespace, name string, options metav1.GetOptions) (*v3.PodSecurityAdmissionConfigurationTemplate, error)
+	List(namespace string, opts metav1.ListOptions) (*v3.PodSecurityAdmissionConfigurationTemplateList, error)
+	Watch(namespace string, opts metav1.ListOptions) (watch.Interface, error)
+	Patch(namespace, name string, pt types.PatchType, data []byte, subresources ...string) (result *v3.PodSecurityAdmissionConfigurationTemplate, err error)
+}
+
+type PodSecurityAdmissionConfigurationTemplateCache interface {
+	Get(namespace, name string) (*v3.PodSecurityAdmissionConfigurationTemplate, error)
```

Comment:
@HarrisonWAffel 
I just realized that `PodSecurityAdmissionConfigurationTemplate` should be a **non-namespaced** resource type. Is it right? 😄 

The fix should be in rancher/rancher: we need to add `// +genclient:nonNamespaced` to where the struct is defined. 

example: https://github.com/jiaqiluo/rancher/blob/ae44c398377516fcd150b8fc2a81a9fd6d4d8fb5/pkg/apis/management.cattle.io/v3/authz_types.go#L128-L128

Reply:
This struct actually isn't needed to accomplish this validation, so it has been removed